index.js

```diff
@@ -1,18 +1,146 @@
+        userGuess = await ask("Too high! Guess again: ");
+      } else if (userGuess < compPick) {
+        userGuess = await ask("Too low! Guess again: ");
+      }
+      userGuess = parseInt(userGuess);
+    }
+    count++;
+    console.log(
+      "You guessed it! It took you " + count + " attempt(s) to get the number!"
+    );
+    playAgain = await ask("Do you want to play again? " + "(y/n):");
+
+    if (playAgain === "n") {
+      console.log("Goodbye!");
+      process.exit();
+    } else if (playAgain === "y") {
+      humanGuessstart();
+    }
+  }
+}
+
+/*--------------Global-------------*/
+
+/*Function*/
+
+//Modify Guess Range
+function randomNumber(min, max) {
+  let range = max - min + 1;
+  return min + Math.floor(Math.random() * range);
+}
+
+//Make it Smarter
+function smartGuess(min, max) {
+  return Math.floor((min + max) / 2);
+}
+//computerGuessstart();
+
+async function computerGuessstart() {
+  console.log(
+    "Let's play a game where you (human) make up a number and set the upper limit, then I (computer) try to guess it."
+  );
+
+  let secretNumber = await ask(
+    "What is your secret number?\nI won't peek, I promise...\n"
+  );
+  secretNumber = parseInt(secretNumber);
+  max = await ask("What is the upper limit of your range?");
+  max = parseInt(max);
+
+  let compGuess = smartGuess(min, max);
+  let answer = await ask("Is your number? " + compGuess + " (y/n):");
+
+  //Let the Computer Win
 
-start();
+  if (answer === "y") {
+    console.log("You rock!");
+    process.exit();
+  } else {
+    while (answer !== "y") {
+      let hiLow = await ask("Is it higher or lower? (h/l):");
 
-async function start() {
-  console.log("Let's play a game where you (human) make up a number and I (computer) try to guess it.")
-  let secretNumber = await ask("What is your secret number?\nI won't peek, I promise...\n");
-  console.log('You entered: ' + secretNumber);
-  // Now try and complete the program.
+      // Modify your guess range
+      if (hiLow === "h" && secretNumber > compGuess) {
+        min = compGuess + 1;
+        compGuess = smartGuess(min, max);
+      } else if (hiLow === "l" && secretNumber < compGuess) {
+        max = compGuess - 1;
+        compGuess = smartGuess(min, max);
+      } else if ((hiLow = "h" && secretNumber < compGuess)) {
+        console.log(
+          "You said it was lower than " +
+            compGuess +
+            ", so it cant also be higher!" +
+            "YOU ARE CHEATING...GAME OVER!"
+        );
+        process.exit();
+      } else if ((hiLow = "l" && secretNumber > compGuess)) {
+        console.log(
+          "You said it was higher than " +
+            compGuess +
+            ", so it cant also be lower! " +
+            "YOU ARE CHEATING...GAME OVER!"
+        );
+        process.exit();
+      }
+      answer = await ask("Is your number" + " " + compGuess + "?" + "(y/n):");
+    }
+  }
+  console.log("I guessed it");
   process.exit();
```

Comment:
You could add some code here that asks if you'd like to play this game again and or if you'd like the play the other game also